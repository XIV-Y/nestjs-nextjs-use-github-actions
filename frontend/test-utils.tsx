import { ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) {
  return render(ui, { ...options });
}

export * from "@testing-library/react";
export { customRender as render };
