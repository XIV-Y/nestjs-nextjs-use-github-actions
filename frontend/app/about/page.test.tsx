import { describe, it, expect } from "vitest";
import { render, screen } from "../../test-utils";
import About from "./page";

describe("About", () => {
  it("Aboutページの見出しが正しく表示されること", () => {
    render(<About />);

    const heading = screen.getByTestId("about-heading");

    expect(heading).toBeInTheDocument();
    expect(heading.textContent).toBe("プロジェクトについて");
  });

  it("説明文が表示されていること", () => {
    render(<About />);

    const description = screen.getByTestId("about-description");

    expect(description).toBeInTheDocument();
    expect(description.textContent).toContain("Next.js 15");
    expect(description.textContent).toContain("App Router");
  });

  it("技術スタックのセクションが表示されていること", () => {
    render(<About />);

    expect(screen.getByText("技術スタック")).toBeInTheDocument();
    expect(screen.getByText("Next.js 15 (App Router)")).toBeInTheDocument();
    expect(screen.getByText("React 19")).toBeInTheDocument();
    expect(screen.getByText("TypeScript")).toBeInTheDocument();
    expect(screen.getByText("Tailwind CSS")).toBeInTheDocument();
    expect(screen.getByText("Vitest + Testing Library")).toBeInTheDocument();
  });

  it("始め方のセクションに適切なコマンドが含まれていること", () => {
    render(<About />);

    const setupSection = screen.getByText("始め方");

    expect(setupSection).toBeInTheDocument();

    const codeBlock = document.querySelector("pre");

    expect(codeBlock).toBeInTheDocument();
    expect(codeBlock?.textContent).toContain("npm install");
    expect(codeBlock?.textContent).toContain("npm run dev");
    expect(codeBlock?.textContent).toContain("npm test");
  });

  it("Aboutページのスナップショットが一致すること", () => {
    const { container } = render(<About />);

    expect(container).toMatchSnapshot();
  });
});
