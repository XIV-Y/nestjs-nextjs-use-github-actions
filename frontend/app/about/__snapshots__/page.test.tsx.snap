// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`About > Aboutページのスナップショットが一致すること 1`] = `
<div>
  <div
    class="space-y-8"
  >
    <section
      class="bg-gradient-to-r from-green-500 to-teal-600 text-white p-12 rounded-xl shadow-lg"
    >
      <h1
        class="text-4xl font-bold mb-4"
        data-testid="about-heading"
      >
        プロジェクトについて
      </h1>
      <p
        class="text-xl mb-6"
      >
        Next.js と Vitest を組み合わせた最新のフロントエンド開発環境
      </p>
    </section>
    <div
      class="bg-white p-8 rounded-lg shadow-md mt-8"
    >
      <h2
        class="text-2xl font-bold text-gray-800 mb-6"
      >
        プロジェクトの目的
      </h2>
      <p
        class="text-gray-600 mb-6"
        data-testid="about-description"
      >
        このプロジェクトは、Next.js 15 の新しいApp Routerアーキテクチャを使用した最新のReactアプリケーションを構築し、 Vitestを用いた効率的なテスト方法を実演することを目的としています。Tailwind CSSを使用して、 モダンで美しいUIを簡単に構築できることも示しています。
      </p>
      <div
        class="grid md:grid-cols-2 gap-8 mt-8"
      >
        <div
          class="bg-gray-50 p-6 rounded-lg"
        >
          <h3
            class="text-xl font-semibold text-gray-800 mb-3"
          >
            技術スタック
          </h3>
          <ul
            class="space-y-2 text-gray-600"
          >
            <li
              class="flex items-center"
            >
              <span
                class="text-green-500 mr-2"
              >
                ●
              </span>
               Next.js 15 (App Router)
            </li>
            <li
              class="flex items-center"
            >
              <span
                class="text-green-500 mr-2"
              >
                ●
              </span>
               React 19
            </li>
            <li
              class="flex items-center"
            >
              <span
                class="text-green-500 mr-2"
              >
                ●
              </span>
               TypeScript
            </li>
            <li
              class="flex items-center"
            >
              <span
                class="text-green-500 mr-2"
              >
                ●
              </span>
               Tailwind CSS
            </li>
            <li
              class="flex items-center"
            >
              <span
                class="text-green-500 mr-2"
              >
                ●
              </span>
               Vitest + Testing Library
            </li>
          </ul>
        </div>
        <div
          class="bg-gray-50 p-6 rounded-lg"
        >
          <h3
            class="text-xl font-semibold text-gray-800 mb-3"
          >
            実装された機能
          </h3>
          <ul
            class="space-y-2 text-gray-600"
          >
            <li
              class="flex items-center"
            >
              <span
                class="text-green-500 mr-2"
              >
                ●
              </span>
               
              レスポンシブデザイン
            </li>
            <li
              class="flex items-center"
            >
              <span
                class="text-green-500 mr-2"
              >
                ●
              </span>
               
              コンポーネントの単体テスト
            </li>
            <li
              class="flex items-center"
            >
              <span
                class="text-green-500 mr-2"
              >
                ●
              </span>
               
              ページのレンダリングテスト
            </li>
            <li
              class="flex items-center"
            >
              <span
                class="text-green-500 mr-2"
              >
                ●
              </span>
               
              モダンなUI/UXデザイン
            </li>
          </ul>
        </div>
      </div>
      <div
        class="mt-8 p-6 bg-blue-50 rounded-lg border border-blue-100"
      >
        <h3
          class="text-xl font-semibold text-blue-800 mb-3"
        >
          始め方
        </h3>
        <pre
          class="bg-gray-800 text-gray-100 p-4 rounded-md overflow-x-auto"
        >
          <code>
            git clone https://github.com/example/nextjs-vitest-example.git
            <br />
            cd nextjs-vitest-example
            <br />
            npm install
            <br />
            npm run dev # 開発サーバーの起動
            <br />
            npm test # テストの実行
          </code>
        </pre>
      </div>
    </div>
  </div>
</div>
`;
